src/main/java/com/google/devtools/build/lib/bazel/repository/skylark/SkylarkRepositoryContext.java

```diff
@@ -487,6 +492,32 @@ private void validateExecuteArguments(Sequence<?> arguments) throws EvalExceptio
     }
   }
 
+  /** Returns the command line arguments as a string for display in the profiler. */
+  private static String profileArgsDesc(String method, List<String> args) {
+    StringBuilder b = new StringBuilder();
+    b.append(method).append(": ");
+
+    String sep = "";
+    for (String arg : args) {
+      int appendLen = arg.length() + sep.length();
+      int remainingLen = MAX_PROFILE_ARGS_LEN - b.length();
+
+      if (appendLen <= remainingLen) {
+        b.append(sep);
+        b.append(arg);
+      } else {
+        String shortenedArg = (sep + arg).substring(0, remainingLen);
+        b.append(shortenedArg);
+        b.append("...");
+        break;
+      }
+
+      sep = " ";
```

Comment:
instead of modifying `sep` in the loop, just remove the space at the end of the string in line 498?

Reply:
done